### DISCUSSION TITLE
Revise export of CSV (add "verse text" option) and DOCX (reorganize tags)

### DISCUSSION THREAD

**Discussion by ghost** [ezra-bible-app/ezra-bible-app#990] [Ideas]
Looking through the CSV and DOCX export features, each does part of what I'm look to accomplish, which is to export verse text AND tags in a useful format.

I see that using the CSV export, tags are included but verse text is not (user data only, based on [discussion #476](https://github.com/ezra-bible-app/ezra-bible-app/discussions/476)).

The DOCX export option does include verse text AND tags, but currently it generates a list of tags at the top of the document representing all the tagged verses in the export. When exporting many tags/verses, this tag list becomes very long and loses value. I have resorted to exporting one DOCX document for each tag which is cumbersome.

With that said, either of two things would be helpful:
- Extend the functionality of the existing CSV "export user data" to have an option to include the verse text associated with tagged verses in a new separate column. 
- OR, revise the existing DOCX export document to have more meaningful verse/tag display. Instead of the tag list at the top of the document, add the relevant tag(s) for each verse in brackets beside each verse reference ( e.g. John 1:1 [tag name(s) here] ), similar to the verse/tag display in the app's main window.

Other nice to haves would be:
- Add Bible translation name (e.g. KJV, NASB) to the top of the DOCX export document.
- Add verse/tag grouping options in the DOCX export document. When selecting the "Export to text document" button, a dropdown with optional checkboxes:
  - Option 1 (checked by default) - Export with verse text grouped by Bible book (export document has this currently) and add tags next to each verse (as mentioned above).
  - Option 2 - Export with verse text grouped by tag. Of course, this option will result multi-tag verses being added multiple times in the DOCX export document, but that would be the desired outcome.

And I just want to say thank for this app. It's wonderful and has taken the top spot among my Bible apps for the tagging feature alone. Also, having those tags "stick" to each verse regardless of the Bible translation I'm viewing is immensely powerful. Other apps assign verse bookmarks to only one Bible translation which has always irked me, but this app does it right.

**Comment by tobias-klein** [comment]
Hi @kaitaia!

Thanks for your positive feedback and these ideas.

Of all the things mentioned, the point about extending the CSV export with additional verse text columns is probably the easiest to realize.

However, one additional aspect needs to be considered here and that is the *copyright* terms of Bible translations. When exporting *all* user data the amount of verses is potentially very high. This may be in conflict with some of the copyright terms out there (limiting the amount of verses).

What we could do in any case is show the same copyright dialog as for the word export. And on top of that we could consider to limit the user data export including Bible text to translations that are copyright-free.

Your ideas for the Word export are nice. Maybe @zhuiks (developer of the current word export) would like to comment here?
